src/DurableTask.Netherite/StorageLayer/Faster/FasterStorageProvider.cs

```diff
@@ -134,6 +134,8 @@ async Task<bool> IStorageLayer.CreateTaskhubIfNotExistsAsync()
                 this.traceHelper.TraceProgress($"Using existing blob container at {this.cloudBlobContainer.Result.Uri}");
             }
 
+            // TODO: Add validation to check if customer is using a Netherite V1 task hub
```

Comment:
Is this the correct place to add validation to make sure customers are not using Netherite V1 task hubs with Netherite V2 and vice versa? Should we check the `StorageFormat` property for this check? If so, what values can we expect for `StorageFormat`?

Reply:
I feel it might be better to do this validation closer to the `return` statement of this method. In particular, I'd do it in (or after) the `catch` block that reads:

```C#
            catch (StorageException e) when (BlobUtils.BlobAlreadyExists(e))
            {
                // taskhub already exists, possibly because a different node created it faster
                this.traceHelper.TraceProgress("Confirmed existing taskhub");
                return false;
            }
```

In that catch block, we know the taskhub exists. So it may be possible to read it and check some property of it.

Not sure what property to check to determine if the taskhub was created w/ Netherite V1. I'd place a debugger to see what gets loaded at runtime to try to infer what data there can be augmented to include the Netherite version.

Reply:
The validation check already exists. All you have to do is add another line at the end of the `StorageFormatVersion` variable at https://github.com/microsoft/durabletask-netherite/blob/60c7d39e03ec35990f2115efc07ca1d09ceb56e1/src/DurableTask.Netherite/StorageLayer/Faster/AzureBlobs/BlobManager.cs#L152C8-L159C1

We are not sure yet that we need it though. Still trying to get this to be a non-breaking change.

Reply:
Based on our discussion last week, once we get PR https://github.com/microsoft/FASTER/pull/896 merged and published, we should be able to keep this a non-breaking change so we don't need the version check.